lib/assets/javascripts/turbograft/page.coffee

```diff
@@ -25,3 +25,42 @@ Page.refresh = (options = {}, callback) ->
 
 Page.open = ->
   window.open(arguments...)
+
+# Providing hooks for objects to set up destructors:
+onReplaceCallbacks = []
+
+# e.g., Page.onReplace(node, unbindListenersFnc)
+# unbindListenersFnc will be called if the node in question is partially replaced
+# or if a full replace occurs.  It will be called only once
+Page.onReplace = (node, callback) ->
+  throw new Error("Node and callback must both be specified") if !node || !callback
```

Comment:
http://stackoverflow.com/questions/5999998/how-can-i-check-if-a-javascript-variable-is-function-type/6000016#6000016 ?